emulator/src/kreator/preprocessor.rs

```diff
@@ -393,6 +395,42 @@ fn get_labels(code: &Vec<String>) -> Result<HashMap<String, u16>, &'static str>
     Ok(labels)
 }
 
+fn get_opc_by_byte_size() -> (Vec<&'static str>, Vec<&'static str>, Vec<&'static str>) {
+    (vec![
+        "NOP", "STAX", "INX", "INR", "DCR", "RLC", "DAD", "LDAX", "DCX", "RRC", "RAL", "RAR",
+        "DAA", "STC", "CMC", "HLT", "SBB", "MOV", "ANA", "XRA", "ORA", "CMP", "RNZ", "POP",
+        "PUSH", "RST", "RZ", "RET", "RNC", "RC", "RPO", "XTHL", "RPE", "PCHL", "XCHG", "RP",
+        "DI", "RM", "SPHL", "EI", "CMA", "ADD", "ADC", "SUB"
+    ],
+    vec![
+        "MVI", "ADI", "ACI", "OUT", "SUI", "IN", "SBI", "ANI", "XRI", "ORI", "CPI"
+    ],
+    vec![
+        "LXI", "SHLD", "LHLD", "STA", "LDA", "JNZ", "JMP", "CNZ", "JZ", "CZ", "CALL", "JNC",
+        "CNC", "JC", "CC", "JPO", "CPO", "JPE", "CPE", "JP", "CP", "JM", "CM"
+    ])
+}
+
+fn get_byte_amount_of_line(line: &String) -> i32 {
+    let (one_byte_labels, two_byte_labels, three_byte_labels) = get_opc_by_byte_size();
+    for opc in &one_byte_labels {
```

Comment:
Possibly you could first get the first wird of the line (the opcode) and then just call x_byte_labels.contains(opcode) or something.